tests/sentry/workflow_engine/processors/test_detector.py

```
@@ -62,9 +79,57 @@ class HandlerStateGroupType(GroupType):
         self.handler_type = HandlerGroupType
         self.handler_state_type = HandlerStateGroupType
 
+    def tearDown(self):
+        super().tearDown()
+        self.sm_comp_patcher.__exit__(None, None, None)
+
+    def create_detector_and_conditions(self, type: str | None = None):
```

Ruff (0.16.2)

[error] Function argument `type` is shadowing a Python builtin

(A002)